### PR TITLE
starting USSD endpoint and starting models refactor

### PR DESCRIPTION
**Pull request by wangfowen**
super barebones USSD endpoint just to get started, and extracted out IpAddress and utils (based on http://exploreflask.com/en/latest/organizing.html) from models.py to start making it not just one huge file

am I doing this stuff right? what's the general work flow for making sure i'm not breaking stuff? i currently just load the admin dashboard then click around and see if anything crashes... doesn't seem optimal

### CODE REVIEW

File: app/server/api/ussd_api.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        Method: POST
+
+        @params sessionId - USSD session ID
+        @params phoneNumber - MSISDN initiating the USSD request
+        @params serviceCode - USSD code dialled to access the service eg. *701#
+        @params text - User input. The initial input is blank
+
+        Method returns a text response to be displayed to the user's screen
+        All responses mid-session begin with CON. All responses that terminate the session begin with the word END
+    """
```
Comment (wangfowen) [inline app/server/api/ussd_api.py]: public API so I don't put a `required_auth` right?

File: app/server/models/ip_address.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from server import db, sentry, celery_app
```
Comment (wangfowen) [inline app/server/models/ip_address.py]: extracted out of models.py

File: app/server/models/utils.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import datetime
```
Comment (wangfowen) [inline app/server/models/utils.py]: extracted out of models.py

File: app/server/api/ussd_api.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            text = "test"
+        else:
+            #TODO: render plain: UssdMenu.find_by(name: 'exit_invalid_request').display_text
+            text = "test2"
+
+        return make_response(text, 200)
+
+
+ussd_blueprint.add_url_rule(
+    '/v1/ussd/kenya',
```
Comment (wangfowen) [inline app/server/api/ussd_api.py]: do I need to put v1 or is there already some prepending of api/v1, etc?

Comment (wangfowen) [reply app/server/api/ussd_api.py]: and looks like if I curl the endpoint I get 404:
```
127.0.0.1 - - [03/Oct/2019 00:17:29] "POST /api/v1/ussd/kenya HTTP/1.1" 404 -
```
am I missing something?

Comment (wangfowen) [inline app/server/api/ussd_api.py]: if i curl the endpoint i get 404:

```
127.0.0.1 - - [03/Oct/2019 00:17:29] "POST /api/v1/ussd/kenya HTTP/1.1" 404 -
```

am I missing something?

**Comment by enjeyw** [comment]
> i currently just load the admin dashboard then click around and see if anything crashes... doesn't seem optimal

unfortunately that's probably the best way _for now_, as test coverage isn't quite there yet. A few more core tests and i think we'll be able to swap to running test watchers... does that seem right to u?

**Comment by enjeyw** [comment]
> models.py to start making it not just one huge file

Ur structure is looking _soooo_ much cleaner. hopefully it doesn't kill the python imports lol

File: app/server/api/ussd_api.py (hunk 1 above)
Comment (enjeyw) [reply app/server/api/ussd_api.py]: that doesn't sound right. what stops a malicious actor just POSTING with a fake session_id or phone number? Does Africa's talking have some sort of field to supply auth headers?

**Comment by enjeyw** [comment]
CI/CD pipeline tests are failing, but i don't think that a problem with this specific PR. I'll look over today.

**Pull request merged**